sa/sa.go

```diff
@@ -354,6 +354,18 @@ func (ssa *SQLStorageAuthority) CountCertificatesByNames(ctx context.Context, do
 	return ret, nil
 }
 
+func (ssa *SQLStorageAuthority) CountCertificatesByExactNames(ctx context.Context, domains []string, earliest, latest time.Time) (map[string]int, error) {
+	ret := make(map[string]int, len(domains))
+	for _, domain := range domains {
+		currentCount, err := ssa.countCertificatesByExactName(domain, earliest, latest)
+		if err != nil {
+			return ret, err
+		}
+		ret[domain] = currentCount
```

Comment:
This function, and CountCertificatesByNames, goes to some effort to construct a map, which the server-side wrapper later deconstructs into a list for transmission, which the client-side wrapper later turns back into a map. To reduce duplication, let's remove at least the server side of this. I.e., change these two functions to return a `[]CountByNames_MapElement` directly, and remove the corresponding code in the server wrapper.